internal/services/storage/storage_container_resource.go

```diff
@@ -45,43 +49,44 @@ func storageContainer() *schema.Resource {
 				ForceNew:     true,
 				ValidateFunc: validate.StorageContainerName,
 			},
-
-			// "resource_group_name": commonschema.ResourceGroupName(),
-			// todo property deprecate
-
 			"storage_account_name": {
 				Type:         schema.TypeString,
 				Required:     true,
 				ForceNew:     true,
 				ValidateFunc: validate.StorageAccountName,
 			},
-
 			"container_access_type": {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
 				Default:  "private",
-				// todo validate correctly
-				/*ValidateFunc: validation.StringInSlice([]string{
+				ValidateFunc: validation.StringInSlice([]string{
 					string(containers.Blob),
 					string(containers.Container),
 					"private",
-				}, false),*/
+				}, false),
 			},
-
-			// todo: replace with metadata
-			/*"properties": {
-				Type:     schema.TypeMap,
-				Computed: true,
-				Elem: &schema.Schema{
-					Type: schema.TypeString,
+			"metadata": {
```

Comment:
Same suggestion as below about Flatten/expand Metadata in validate `metadata.go` just to be consistent with the rest of the project

Reply:
ready!